### PR TITLE
Add a new recipe to run the Sail tests manually via GH Actions

### PR DESCRIPTION
**Pull request by rpsene**
This PR adds the capability to run the tests manually without depending on a new change to trigger it.

### CODE REVIEW

**Comment by jrtc27** [comment]
Why? If the model hasn't changed the tests really shouldn't. On the very rare occasion that you need a manual run for some unusual set of circumstances you can just open a dummy PR, but I don't see how this is very useful.

**Comment by jrtc27** [comment]
This certainly shouldn't be duplicating so much code if you really do want to add it for some reason. Reuse the existing code and gate steps that are specific to PRs.

**Review by jrtc27 — CHANGES_REQUESTED** [review]
Unjustified and unnecessarily duplicates code

**Comment by rpsene** [comment]
@jrtc27 thanks for the feedback, PR changed. Initially I was not doing executing all same steps.

File: .github/workflows/compile.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: CI
 
-on: [push, pull_request]
+on: [push, pull_request,workflow_dispatch]
```
Comment (jrtc27) [inline .github/workflows/compile.yml:3]:
```suggestion
on: [push, pull_request, workflow_dispatch]
```
...

Comment (rpsene) [reply .github/workflows/compile.yml]: fixed in the original PR.

**Review by billmcspadden-riscv — APPROVED** [review]
This looks fine to me.

**Comment by jrtc27** [comment]
The minute notes just say "2 use cases" and that this was approved. That is totally useless for anyone not in the meetings. Please justify somewhere, especially when I have explicitly objected to this change and asked for the motivation a week ago.

**Pull request merged**